conversion/hcl/compute_instance.go

```diff
@@ -0,0 +1,417 @@
+package hcl
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/terraform-google-conversion/v2/helper"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"google.golang.org/api/compute/v1"
+)
+
+const ComputeInstanceAssetType string = "compute.googleapis.com/Instance"
+
+func NewComputeInstanceConverter() *Converter {
+	return &Converter{
+		TFResourceName: "google_compute_instance",
+		Convert:        convertComputeInstance,
+	}
+}
+
+func convertComputeInstance(asset *Asset) (string, map[string]interface{}, error) {
+	if asset == nil || asset.Resource == nil || asset.Resource.Data == nil {
+		return "", nil, fmt.Errorf("asset does not provide enough data for conversion")
+	}
+	var instance *compute.Instance
+	if err := helper.DecodeJSON(asset.Resource.Data, &instance); err != nil {
+		return "", nil, err
+	}
+
+	bootDisks, scratchDisks, attachedDisks := convertDisks(instance.Disks)
```

Comment:
Looking back, the diff includes:
- found one flatten function includes using schema.Set to build fields, which needs to introduce other functions like hash etc, and this makes the unmarshal/marshal fail, since schema.Set is not implementing the json marshal interface. 
- found a few flatten function takes config as input parameter, and will send API call to google service, eg. `func flattenBootDisk(d *schema.ResourceData, disk *compute.AttachedDisk, config *Config) []map[string]interface{}`

Reply:
this should be fine for now; we might reconsider when we switch to generating this implementation from magic modules.